assets/src/domain/eventEditor/ui/datetimes/datesList/actionsMenu/dropdown/DateMainMenu.tsx

```diff
@@ -1,29 +1,59 @@
-import React from 'react';
+import React, { useCallback, useRef } from 'react';
+import { useDisclosure } from '@chakra-ui/core';
 import { __ } from '@wordpress/i18n';
 
+import { AlertDialog } from '@infraUI/display';
+import { Button, ButtonType } from '@application/ui/input';
 import { DropdownMenu, DropdownToggleProps } from '@application/ui/layout';
+import { isTrashed } from '@sharedServices/predicates';
+import { useDatetimeItem } from '@edtrServices/apollo';
+import { useDatetimeMutator } from '@edtrServices/apollo/mutations';
 
 import CopyDate from './CopyDate';
 import EditDate from './EditDate';
 import TrashDate from './TrashDate';
-import { useDatetimeItem } from '@edtrServices/apollo';
 
 import { DateMainMenuProps } from './types';
 
 const DateMainMenu: React.FC<DateMainMenuProps> = ({ datetime: entity }) => {
+	const { isOpen, onOpen, onClose } = useDisclosure();
+	const cancelRef = useRef();
 	// Make sure to subscribe to Apollo cache
 	// to avoid stale data
 	const datetime = useDatetimeItem({ id: entity.id });
+	const { id } = datetime;
+	const trashed = isTrashed(datetime);
+	const { deleteEntity } = useDatetimeMutator(datetime.id);
+	const onClickHandler = useCallback((onClick) => {
+		if (typeof onClick === 'function') {
+			onClick();
+		}
+		deleteEntity({ id, deletePermanently: trashed });
+		onClose();
+	}, []);
+
 	const toggleProps: DropdownToggleProps = {
 		tooltip: __('event date main menu'),
 	};
 
 	return (
-		<DropdownMenu toggleProps={toggleProps}>
-			<EditDate datetime={datetime} />
-			<CopyDate datetime={datetime} />
-			<TrashDate datetime={datetime} />
-		</DropdownMenu>
+		<>
+			<DropdownMenu toggleProps={toggleProps}>
+				<EditDate datetime={datetime} />
+				<CopyDate datetime={datetime} />
+				<TrashDate datetime={datetime} onClickHandler={onClickHandler} onOpen={onOpen} />
+			</DropdownMenu>
+			<AlertDialog
+				cancelButton={<Button buttonText={__('No')} ref={cancelRef} onClick={onClose} />}
+				header={__('Are you sure you want to delete this?')}
+				isOpen={isOpen}
+				leastDestructiveRef={cancelRef}
+				okButton={
+					<Button buttonText={__('Yes')} buttonType={ButtonType.ACCENT} onClick={onClickHandler} ml={3} />
+				}
+				onClose={onClose}
+			/>
```

Comment:
Because `MenuItem` should be direct child of `MenuList`, see here: https://chakra-ui.com/menu#import
I can't render this, nor `ConfirmDelete` as a wrapper of `MenuItem`.

Reply:
in action:

![Peek 2020-05-05 00-51](https://user-images.githubusercontent.com/1477453/81017206-996daa00-8e6a-11ea-8961-c957550dee01.gif)

Reply:
hmmm... ya tricky... 
would be much better having this logic colocated with the component that spawns it...

Does the `AlertDialog` have to be rendered outside of `DropdownMenu` in order to work?
More specifically, do you know if it will work if it is rendered as a child of `MenuItem` within the `DropdownMenuItem` component?

Because off the top of my head, I'm wondering if something like the following would work:



###### /application/ui/inputs/confirmationDialog/useConfirmationDialog .tsx
```js
const useConfirmationDialog = ({ confirmText, onConfirm, ...props }) => {
	const cancelRef = useRef();
	const { isOpen, onOpen, onClose } = useDisclosure();

	const noButtonText = props.noButtonText || __('No');
	const yesButtonText = props.yesButtonText || __('Yes');

	const onClickHandler = useCallback((onConfirm) => {
		if (typeof onConfirm=== 'function') {
			onConfirm();
		}
		onClose();
	}, []);

	const cancelButton = (
		<Button
			buttonText={ noButtonText }
			ref={ cancelRef }
			onClick={ onClose }
		/>
	);

	const okButton = (
		< Button
			buttonText={ yesButtonText }
			buttonType={ ButtonType.ACCENT }
			onClick={ onClickHandler }
			ml={ 3 }
		/>
	);

	const confirmationDialog = (
		<AlertDialog
			cancelButton={ cancelButton }
			header={ confirmText }
			isOpen={ isOpen }
			leastDestructiveRef={ cancelRef }
			okButton={ okButton }
			onClose={ onClose }
		/>
	);

	return {
		confirmationDialog,
		onOpen
	};
};
```


###### /application/ui/layout/entityActionsMenu/entityMenuItems/Trash.tsx
```js
const Trash: React.FC<MenuItemProps> = ({ onClick, ...props }) => {
	const title = props.title || __('trash');
	const confirmText = props.confirmText || __('Are you sure you want to delete this?');

	const { confirmationDialog, onOpen } = useConfirmationDialog({
		confirmText,
		onConfirm: onClick,
	});

	return (
		<DropdownMenuItem
			{ ...props }
			extra={ confirmationDialog }
			icon={ TrashIcon }
			onClick={ onOpen }
			title={ title }
		/>
	);
};
```


###### /application/ui/layout/dropdownMenu/DropdownMenuItem.tsx
```js
const DropdownMenuItem: React.FC<DropdownMenuItemProps> = ({ extra, icon: Icon, title, ...props }) => {
	const className = classNames('ee-dropdown-menu__item', props.className);
	return (
		<MenuItem {...props} className={className} role='menuitem'>
			{Icon && <Icon />}
			<span>{title}</span>
			{ extra }
		</MenuItem>
	);
};
```

which would then make that logic automatically available to any code using a trash button, and not require anything more to be done for tickets, unless you wanted to customize the confirmation alert text for dates and tickets. Would also make the confirmation dialog available for other uses.

Reply:
I've moved the `AlertDialog` logic within `MenuItem`, w/o extraction to see if that works and as can be seen below and in the last commit, it won't work.
As I understand when menu is closed the `MenuItem` is removed from the DOM tree and so is the `AlertDialog`, that is why in the previous PR I've placed that component at the same level as `DropdownMenu`.
![Peek 2020-05-05 17-27](https://user-images.githubusercontent.com/1477453/81079359-23b01f80-8ef8-11ea-8e8c-35ac1d61fa7a.gif)

Reply:
other stuff I tried and didn't worked:
I've tried to use `useMenuContext` to be able to execute `closeMenu` upon alert dialog confirmation and although it is exported here, I couldn't use it directly: https://github.com/chakra-ui/chakra-ui/blob/master/packages/chakra-ui/src/Menu/index.js#L169